src/validate.rs

```diff
@@ -900,6 +890,33 @@ fn validate_repos(data: &Data, errors: &mut Vec<String>) {
     });
 }
 
+fn check_team_access(
+    data: &Data,
+    all_github_teams: &HashSet<(String, String)>,
+    repo: &Repo,
+    team_name: &str,
+) -> anyhow::Result<()> {
+    let team = data
+        .team(team_name)
+        .with_context(|| format!("The file 'teams/{team_name}.toml' does not exist."))?;
+    let github_teams = team.github_teams(data)?;
+    let mut team_found = false;
+    for github_team in &github_teams {
+        if all_github_teams.contains(&(repo.org.clone(), github_team.name.to_string())) {
+            team_found = true;
+        } else if data.is_team_archived(github_team.name, &repo.org) {
```

Comment:
The function name is confusing, it means `is_github_team_archived`, right? But I don't think we need to check GH team archival status after this redesign. We should just check if the team itself is archived or not, and that's it. GH teams don't have any additional archival status.